graphql/context.go

```diff
@@ -29,12 +29,14 @@ type RequestContext struct {
 	Errors   gqlerror.List
 }
 
-func DefaultResolverMiddleware(ctx context.Context, next Resolver) (res interface{}, err error) {
-	return next(ctx)
+func DefaultResolverMiddleware(ctx context.Context, next Resolver) (context.Context, interface{}, error) {
+	res, err := next(ctx)
```

Comment:
Does this interface mismatch imply resolvers cant change their childs context either? Should we be thinking a little broader than just directives?

Reply:
It is an interesting point.
Does resolvers and directives need to modify context?
For Resolver, I can not make an idea that use case.
For directive, It doesn't have tree structure, because it doesn't need to modify context. I think directive middleware may be good.